CONTRIBUTING.md

```diff
@@ -19,78 +19,117 @@ If you're new to open source in general, check out [GitHub's open source intro g
+7. **Create a Pull Request** from your fork to the main repository
 
-Another good page, if you want to interactively test out your changes, is `http://localhost:8000/test/demo/basic.html`.
+Please provide a clear description in your pull request that explains:
+- What the change does
+- Why it's needed
+- How it's been tested
+- Any related issues (use "Fixes #123" or "Relates to #123" syntax)
 
 
-> Note: corresponding npm commands for the above are:
->  - npm run test
->  - npm run test -- --module="navigator"
->  - npm run dev
+### Troubleshooting
 
-You can also get a report of the tests' code coverage:
+#### Common Testing Issues
 
-    grunt coverage
+- **Tests failing in headless mode only**: This can sometimes be related to timing issues. Try running with `--no-headless` flag for debugging
+- **Browser compatibility issues**: Ensure you're testing with the supported browsers
+- **Timeout errors**: May indicate performance issues or blocking operations in the code
 
-The report shows up at `coverage/html/index.html` viewable in a browser.
+### Development Environment Tips
 
-### Installing from forked Github repo/branch
+- **VS Code users**: Install the EditorConfig extension for automatic code style compliance
+- **Browser DevTools**: Use the Network panel to debug tile loading issues
+- **Debugging**: Open your browser's developer tools (F12) when testing locally
+- **Test isolation**: When debugging a specific test, use `?module=moduleName` in the URL query parameters
 
-This project is now compatible with direct installation of forked Github repos/branches via npm/yarn (possible because of the new [prepare](https://docs.npmjs.com/misc/scripts) command).  This enables quick testing of a bugfix or feature addition via a forked repo.  In order to do this:
+#### Common Build Issues
 
-1. Install the Grunt command line runner (if you haven't already); on the command line, run `npm install -g grunt-cli` (or `yarn global add grunt-cli`)
-1. Remove any currently installed openseadragon package via `npm uninstall openseadragon` or `yarn remove openseadragon`
-1. Add the specific forked repo/branch by running `npm install git://github.com/username/openseadragon.git#branch-name` or `yarn add git://github.com/username/openseadragon.git#branch-name`. Make sure to replace username and branch-name with proper targets.
+- **Missing dependencies**: Run `npm install` to update all dependencies
+- **Grunt errors**: Ensure you're using a compatible Node.js version (check package.json)
+- **Build failing silently**: Check for JavaScript errors in your browser's console
 
-During installation, the package should be correctly built via grunt and can then be used via `import Openseadragon from 'openseadragon'` or `var Openseadragon = require('openseadragon')` statements as if the official package were installed.
+### Developer Resources
+
+- **API Documentation**: Generated documentation is available in the doc folder after build
```

Comment:
This is not true. In order to build the documentation you need to use the https://github.com/openseadragon/site-build/ repository. You can see the currently deployed documentation at https://openseadragon.github.io/docs/.